napalm_ios/ios.py

```diff
@@ -1134,129 +1136,225 @@ def bgp_time_conversion(bgp_uptime):
+                 'record': False},
+                {'regexp': re.compile(r'^\s+\d+\s+(?P<remote_as>\d+)(\s+\S+){5}\s+'
+                                      r'(?P<uptime>(never)|\d+\S+)\s+(?P<accepted_prefixes>\d+)'),
+                 'record': True},
+                {'regexp': re.compile(r'^\s+\d+\s+(?P<remote_as>\d+)(\s+\S+){5}\s+'
+                                      r'(?P<uptime>(never)|\d+\S+)\s+(?P<state>\D.*)'),
+                 'record': True}
+            ],
+            'no_fill_fields': ['accepted_prefixes', 'state']
+        }
+        parse_neighbors = {
+            'patterns': [
+                {'regexp': re.compile(r'^BGP neighbor is '
+                                      r'(?P<remote_addr>(\d+(\.\d+){3})|([0-9a-fA-F:]{2,40})),'
+                                      r'\s+remote AS (?P<remote_as>\d+).*'),
+                 'record': False},
+                {'regexp': re.compile(r'^\s+Description: (?P<description>.+)'),
+                 'record': False},
+                {'regexp': re.compile(r'^\s+BGP version \d+, remote router ID '
+                                      r'(?P<remote_id>\d+(\.\d+){3})'),
+                 'record': False},
+                {'regexp': re.compile(r'^\s+For address family: (?P<afi>\S+) (?P<safi>\S+)'),
+                 'record': False},
+                {'regexp': re.compile(r'^\s+Prefixes Current:\s+(?P<sent_prefixes>\d+)\s+'
+                                      r'(?P<accepted_prefixes>\d+).*'),
+                 'record': False},
+                {'regexp': re.compile(r'^\s+Saved (soft-reconfig):.+(?P<received_prefixes>\d+).*'),
+                 'record': True},
+                {'regexp': re.compile(r'^\s+Local Policy Denied Prefixes:.+'),
+                 'record': True}
+            ],
+            'no_fill_fields': ['received_prefixes']
+        }
+        # parse outputs into a list of dicts
+        summary_data = []
+        summary_data_entry = {}
+        for line in summary_output.splitlines():
+            for item in parse_summary['patterns']:
+                match = item['regexp'].match(line)
+                if match:
+                    summary_data_entry.update(match.groupdict())
+                    if item['record']:
+                        summary_data.append(copy.deepcopy(summary_data_entry))
+                        for field in parse_summary['no_fill_fields']:
+                            try:
+                                del summary_data_entry[field]
+                            except KeyError:
+                                pass
+                    break
+        neighbor_data = []
+        neighbor_data_entry = {}
+        for line in neighbor_output.splitlines():
+            for item in parse_neighbors['patterns']:
+                match = item['regexp'].match(line)
+                if match:
+                    neighbor_data_entry.update(match.groupdict())
+                    if item['record']:
```

Comment:
i'm basically re-creating a single-state implementation of `textfsm` ;-)
`record` indicates a logical row break, so a match against a row with `record == True` will trigger the insertion of a new row.
when that occurs, keys listed in `no_fill_fields` get removed from the temp dict before the next pass, so that the values don't get copied across rows. fields that are not listed, and are not overridden on the next pass get filled down (e.g. local_as gets set once at the top of the output, and then filled down in each row).
comments added to make this clearer.